target/linux/ath79/dts/qca9533_qca_ap143.dts

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca953x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "qca,ap143", "qca,qca9533";
+	model = "Qualcomm Atheros AP143 reference board";
+
+	aliases {
+		label-mac-device = &wmac;
+		led-boot = &led_wps;
+		led-failsafe = &led_wps;
+		led-running = &led_wps;
+		led-upgrade = &led_wps;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wan {
+			label = "green:wan";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "green:wlan";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		lan1 {
+			label = "green:lan1";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		lan2 {
+			label = "green:lan2";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		lan3 {
+			label = "green:lan3";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		lan4 {
+			label = "green:lan4";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wps: wps {
+			label = "green:wps";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "WPS button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x050000 0x7a0000>;
+			};
+
+			art: partition@ff0000 {
```

Comment:
0x7f0000 or 0xff0000?

Probably we should just add both variants, qca9533_qca_ap143-8m.dts and qca9533_qca_ap143-16m.dts ...

Reply:
Now this one is interesting - the partition table was built up correctly at runtime, which is why I didn't spot this.

[    0.348485] Creating 4 MTD partitions on "spi0.0":
[    0.353459] 0x000000000000-0x000000040000 : "u-boot"
[    0.359814] 0x000000040000-0x000000050000 : "u-boot-env"
[    0.366416] 0x000000050000-0x0000007f0000 : "firmware"
[    0.376892] 2 uimage-fw partitions found on MTD device firmware
[    0.383091] Creating 2 MTD partitions on "firmware":
[    0.388255] 0x000000000000-0x0000001f0000 : "kernel"
[    0.394442] 0x0000001f0000-0x0000007a0000 : "rootfs"
[    0.400659] mtd: device 4 (rootfs) set to be root filesystem
[    0.408685] 1 squashfs-split partitions found on MTD device rootfs
[    0.415172] 0x000000490000-0x0000007a0000 : "rootfs_data"
[    0.421878] 0x0000007f0000-0x000000800000 : "art"

Anyway, I'll fix this.

Reply:
The partition@ff0000 is not functional, it's "just a name". The relevant data is in `reg`. So, if you just want to support the 8m version, simply change to partition@7f0000 ...